### PR TITLE
[action] add GasTipCap and GasFeeCap to AbstractAction

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
EIP-1559 introduced 2 fields: `GasTipCap` and `GasFeeCap`, which can be used by all types of transactions. So add these 2 into `AbstractAction`, which is our common struct for all tx

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by millken — APPROVED** [review]

File: action/actctx.go [hunk 1]
```diff
@@ -54,6 +59,22 @@ func (act *AbstractAction) SetGasPrice(val *big.Int) {
...
+// GasFeeCap returns the gas fee cap
+func (act *AbstractAction) GasFeeCap() *big.Int {
+	if act.gasFeeCap == nil {
+		return act.GasPrice()
+	}
+	return new(big.Int).Set(act.gasFeeCap)
+}
+
 // BasicActionSize returns the basic size of action
 func (act *AbstractAction) BasicActionSize() uint32 {
```
Comment (envestcc) [inline action/actctx.go:79]: the two new fields should be accounting in BasicActionSize()?

File: action/envelope.go [hunk 2]
```diff
@@ -269,8 +225,5 @@ func (elp *envelope) LoadProto(pbAct *iotextypes.ActionCore) error {
 	return nil
 }
 
-// SetNonce sets the nonce value
-func (elp *envelope) SetNonce(n uint64) { elp.nonce = n }
-
 // SetChainID sets the chainID value
 func (elp *envelope) SetChainID(chainID uint32) { elp.chainID = chainID }
```
Comment (envestcc) [inline action/envelope.go:229]: also removed?

**Review by CoderZhi — APPROVED** [review]

File: action/actctx.go (hunk 1 above)
Comment (dustinxie) [reply action/actctx.go:79]: fixed

File: action/envelope.go (hunk 2 above)
Comment (dustinxie) [reply action/envelope.go:229]: `SetChainID ` is not called on `AbstractAction` but called by `Envelope`, so it's fine to keep it here

**Pull request merged**